tests/meta/workflow/test_analysis.py

```diff
@@ -347,50 +399,129 @@ def test_prepare_fastq_files_decompression_needed(
+    sample_id: str,
+):
+    """Tests that are_all_spring_files_present returns False when all files for a case are archived and not retrieved."""
+
+    # GIVEN that the only Spring file belonging to a case has an Archive entry which has retrieved_at not set
+
+    # WHEN checking if all spring files are present
+
+    # THEN the result should be False
+    assert not mip_analysis_api.are_all_spring_files_present(case_id)
+
+
+def test_ensure_files_are_present(
+    mip_analysis_api: MipDNAAnalysisAPI,
+    analysis_store: Store,
+    archived_spring_file,
+    case_id: str,
+    sample_id: str,
+):
+    """Tests that archived files are requested via Miria after the flow cell check has been performed."""
+
+    # GIVEN that the only Spring file belonging to a case has an Archive entry which has retrieved_at not set
+
+    # WHEN ensuring that all files are present
+
+    with mock.patch.object(AnalysisAPI, "ensure_flow_cells_on_disk"), mock.patch.object(
+        SpringArchiveAPI,
+        "retrieve_case",
+    ) as request_submitter:
+        mip_analysis_api.ensure_files_are_present(case_id)
+
+    # THEN the files should have been requested
+    request_submitter.assert_called()
+
+
+@pytest.mark.parametrize("flow_cell_status", [FlowCellStatus.REMOVED, FlowCellStatus.ON_DISK])
+def test_does_any_spring_file_need_to_be_retrieved_flow_cell_status(
+    mip_analysis_api: MipDNAAnalysisAPI,
+    analysis_store: Store,
+    helpers: StoreHelpers,
+    flow_cell_status: str,
+):
+    """Tests that does_any_spring_file_need_to_be_retrieved returns True if one of
+    the flow cells has status 'removed' and False if the status is instead 'ondisk'."""
+
+    # GIVEN a case with a flow cell with 'removed' status
     case: Case = analysis_store.get_cases()[0]
     helpers.add_flow_cell(
         analysis_store,
         flow_cell_name="flowcell_test",
         archived_at=datetime.now(),
         sequencer_type=Sequencers.NOVASEQ,
         samples=analysis_store.get_samples_by_case_id(case.internal_id),
-        status=FlowCellStatus.ON_DISK,
+        status=flow_cell_status,
         date=datetime.now(),
     )
+    # WHEN checking if any files need to be retrieved
 
-    # GIVEN that decompression of spring files is running.
-    with mock.patch.object(
-        PrepareFastqAPI, "is_spring_decompression_needed", return_value=False
-    ), mock.patch.object(
-        PrepareFastqAPI, "is_spring_decompression_running", return_value=True
-    ), mock.patch.object(
-        PrepareFastqAPI, "add_decompressed_fastq_files_to_housekeeper", return_value=None
-    ):
-        with pytest.raises(AnalysisNotReadyError):
-            # WHEN running prepare_fastq_files
-            # THEN an AnalysisNotReadyError should be thrown
-            mip_analysis_api.prepare_fastq_files(case_id=case.internal_id, dry_run=False)
+    # THEN the outcome should be False if the flow cell status is on disk but not otherwise
+    if flow_cell_status == FlowCellStatus.ON_DISK:
```

Comment:
can't you parametrise the output too? so you don't need to do the if statement here

Reply:
like `assert mip_analysis_api.does_any_file_need_to_be_retrieved(case.internal_id) == result`

Reply:
That I can do!